### ISSUE TITLE
Question: Can I add an inputted value onto the find method?

### ISSUE THREAD

**Issue by ryankauk** [Automattic/mongoose#8032]
Hey, just started using mongodb and I'm testing it out. I'm playing around with the possibilities of Dataloader. I want to ensure the position with the dataloader but I'm trying to build a generic dataloader for any mongo query. All I want to do is return back a property for a key I generated on the client side so that when the value is returned I can map this to the key generated for the dataloader. Is this possible with the find method?

I know it's possible with the aggregate function with $addFields, but I would have to do this to every query and would cause performance issues. 

If not possible, do you have any any ideas for another direction I can pursue?

Thanks.

**Comment by vkarpov15** [comment]
Hi, I'm not sure I understand your question, can you please clarify with some code samples? Specifically, this: "All I want to do is return back a property for a key I generated on the client side so that when the value is returned I can map this to the key generated for the dataloader" doesn't quite make sense to me.

Also, by dataloader do you mean [this project](https://github.com/graphql/dataloader)?

**Comment by ryankauk** [comment]
Yes that's the project I'm talking about. I'm just trying to batch the request by using aggregations or combining into one query. I also saw the mongodb batch [method](https://docs.mongodb.com/manual/reference/method/db.collection.initializeOrderedBulkOp/#db.collection.initializeOrderedBulkOp). This might solve my problem, I'll have to look into it a bit more. I'll try to come up with a code sample this week. It's just that with most dataloader situations you have to normalize the return to the proper key associated with the dataloader request. Was just wondering if there's a way I could attach the original dataloader key in the return result of the mongoose request.
